zerver/actions/realm_settings.py

```diff
@@ -426,6 +431,16 @@ def do_set_realm_stream(
         send_event_on_commit(realm, event, active_user_ids(realm.id))
 
 
+def do_set_realm_moderation_request_channel(
+    realm: Realm, stream: Stream | None, stream_id: int, *, acting_user: UserProfile | None
+) -> None:
+    if stream is not None and stream.is_public():
+        raise JsonableError(_("Only private channels are offered for this setting."))
```

Comment:
"Moderation request channel must be private." seems a clearer and simpler for this? I want to avoid the "offered for this setting" framing, and I'm not sure we'll end up reusing the string for anything else.